src/bidmanager.js

```diff
@@ -138,15 +140,56 @@ exports.addBidResponse = function (adUnitCode, bid) {
     }
 
     bid.adserverTargeting = keyValues;
+  }
+
+  function doCallbacksIfNeeded() {
```

Comment:
hah! Good catch. That's because I didn't know what the problem domain was 😊 .

My understanding is that the callbacks are supposed to be executed whenever the auction has ended. Given that:

It makes sense that `addBidToAuction` executes them if all the bids have been received, because there's no sense in waiting if we're not expecting any more bids.

It makes sense that `requestBids` in `prebid.js` [calls them](https://github.com/prebid/Prebid.js/blob/master/src/prebid.js#L412) if the timeout has been reached 

...but those were the only two cases I could think of. These lines seem to be ending the auction based on a third condition... but I have no idea what that condition is, or why it exists. Honestly I would have deleted them... but since the unit tests on this method were sparse, I didn't feel comfortable doing that without checking in with someone who had more context.

Can you shed any light on it?